provers/evm-prover/ARCHITECTURE.md

```diff
@@ -0,0 +1,76 @@
+# EVM Prover
+
+The EVM Prover implements the Prover service for EVM rollups, creating zero-knowledge range proofs for state membership and state transitions. It uses the following SP1 programs to generate these range proofs.
+
+## SP1 Programs
+
+### blevm
+
+The `blevm` SP1 program generates proofs of data inclusion and execution for single EVM blocks.
+
+#### Inputs:
+- `KeccakInclusionToDataRootProofInput`: Data for inclusion proof
+- `EthClientExecutorInput`: Data for EVM execution
+- `celestia_header_hash`: Celestia block header hash
+- `data_hash_bytes`: Hash of the block data
+- `proof_data_hash_to_celestia_hash`: Merkle proof connecting data hash to Celestia hash
```

Comment:
It would help me if these were more descriptive.

For example:
- "Data for inclusion proof" => "Data for inclusion proof. This proof proves that the EVM raw block data was published as a blob and included in a Celestia block"
- "Hash of the block data" => is this EVM block data?

Reply:
Updated.